task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -171,6 +187,56 @@ def _pod_event_watch_loop(self) -> None:
                         "Exception encountered while watching Pod events - restarting watch!")
         logger.debug("Exiting Pod event watcher - stop requested.")
 
+    def _filter_task_configs_in_pods(
+            self,
+            pods: List[V1Pod]
+    ) -> List[Tuple[KubernetesTaskConfig, V1Pod]]:
+        """
+        Called during reconciliation task loop in order to filter the task_configs/pods
+        that are in task_metadata.
+        """
+        v1pods = {pod.metadata.name: pod for pod in pods}
+        task_config_pods = []
+
+        for pod_name, task_metadata in self.task_metadata.items():
+            if pod_name in v1pods:
+                task_config_pod = (task_metadata.task_config, v1pods[pod_name])
+                task_config_pods.append(task_config_pod)
+
+        return task_config_pods
```

Comment:
hmm, I think that we're still not covering the case that jfong mentioned: let's say tron has some sort of outage at the same time as some sort of k8s outage (or human k8s cleanup) - when tron/task_proc restart it's possible that we have some state stored for a previously running pod and k8s no longer has any events/pod metadata.

in this case: we won't return that task_config -> we won't see it in `_filter_task_configs_pods_to_reconcile` -> we'll have an "orphaned" task_config

i think perhaps what we want to do here is maybe something like
```suggestion
        pods_seen = {pod.metadata.name: pod for pod in pods}
        task_config_pods = []

        for pod_name, task_metadata in self.task_metadata.items():
            task_config_pod = (task_metadata.task_config, pods_seen.get(pod_name))
            task_config_pods.append(task_config_pod)

        return task_config_pods
```
and then treat having a `None` for the `V1Pod` entry in the tuple as a LOST event